openedx/core/djangoapps/course_roles/data.py

```diff
@@ -0,0 +1,160 @@
+"""
+Permissions for course roles app.
+
+These are the permissions that can be assigned to a CourseRole which grants access for a course,
+but can also be assigned for org wide course access or instance wide course access.
+They are defined in the database in the course_roles_permission table.
+
+To add a new permission, add a new entry to the CourseRolesPermission enum,
+then and add a new row to the course_roles_permission table in database,
+with a migration.
+
+To remove a permission, remove the entry from the CourseRolesPermission enum,
+and remove the row from the course_roles_permission table in database,
+with a migration.
+
+To change the readable_name or description of a permission, change the
+corresponding entry in CourseRolesPermission enum.
+
+To change the name of a permission, change the corresponding entry on the
+CourseRolesPermission enum, and change the name field of the corresponding
+row in the course_roles_permission table in database, with a migration.
+"""
+from attrs import frozen, field, validators
+from enum import Enum, unique
+
+from django.utils.translation import gettext as _
+
+
+@frozen
+class PermissionData:
+    """
+    Data class for a permission.
+    """
+    name: str = field(validator=validators.instance_of(str))
+    readable_name: str = field(validator=validators.instance_of(str))
+    description: str = field(validator=validators.instance_of(str))
```

Comment:
Nit: I really hope we can lean on mypy to validate types for this.

Reply:
I hope too, I followed the example in [OEP-49](https://open-edx-proposals.readthedocs.io/en/latest/best-practices/oep-0049-django-app-patterns.html#data-py)